From 1de509497826faa0ad84b82f5e2c3d21ee613459 Mon Sep 17 00:00:00 2001
From: Moody Liu <mooodyhunter@outlook.com>
Date: Sat, 13 May 2023 17:39:16 +0100
Subject: [PATCH] riscv64: adjust type definitions

CHAR8 needs to be defined while BOOLEAN should be removed
here to prevent typedef conflicts

Upstream-Status: Backport [https://sourceforge.net/p/gnu-efi/code/ci/1de509497826faa0ad84b82f5e2c3d21ee613459/]
Signed-off-by: Moody Liu <mooodyhunter@outlook.com>
---
 inc/riscv64/efibind.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/inc/riscv64/efibind.h b/inc/riscv64/efibind.h
index 4fdf81d..d8b4f39 100644
--- a/inc/riscv64/efibind.h
+++ b/inc/riscv64/efibind.h
@@ -32,11 +32,9 @@ typedef uint16_t                UINT16;
 typedef int16_t                 INT16;
 typedef uint8_t                 UINT8;
 typedef int8_t                  INT8;
+typedef char                    CHAR8;
 typedef wchar_t                 CHAR16;
 #define WCHAR                   CHAR16
-#ifndef BOOLEAN
-typedef uint8_t                 BOOLEAN;
-#endif
 #undef VOID
 typedef void                    VOID;
 typedef int64_t                 INTN;
-- 
2.41.0

